### ISSUE TITLE
ConsecutiveLiteralAppends: false negative with `this` pointer before consecutive append operation

### ISSUE THREAD

**Issue by mohui1999** [pmd/pmd#4424]
**Affects PMD Version:** 6.55.0

**Rule:** [ConsecutiveLiteralAppends](https://pmd.github.io/latest/pmd_rules_java_performance.html#consecutiveliteralappends)

**Description:**
In this case, it uses consecutive append operations. It should use `this.sb.append("Hello How are you today world");` Thus PMD should report **ConsecutiveLiteralAppends** here but does not.  So it may be a false negative.

**Code Sample demonstrating the issue:**
Maybe a false negative:

```java
public class Foo {
    StringBuffer sb = new StringBuffer(200);
    public void bar() {
        this.sb.append("Hello");                   // NOT report here
        this.sb.append("How are you today world");
    }
}
```

Correctly reported case:

```java
public class Foo {
    StringBuffer sb = new StringBuffer(200);
    public void bar() {
        sb.append("Hello");                      // report here
        sb.append("How are you today world");
    }
}
```

**Expected outcome:**

PMD should report a violation at line 4, but doesn't. This is a false negative.

**Running PMD through:** *[CLI]*

**Comment by mohui1999** [comment]
Fixed in PMD 7.0
